pkg/receiver/jobreceiver/README.md

```diff
@@ -0,0 +1,130 @@
+two ways depending on the configured `type`.
+
+By default, the `type: 'event'` output handler is used. When configured with
+the event output type, command output will be buffered until the process exits,
+at which time the receiver will emit a single structured event capturing the
+command output and optionally metadata about the execution such as exit code
+and duration. This is useful for true monitoring jobs, such as using a
+monitoring plugin not supported natively by OTEL.
+
+When configured with the `log_entries` output type, command output will be
+emitted as distinct log entries line-by-line. This can be used to act as a
+bridge to third party logs and other log events that may not be straightforward
+to access.
+
+[stanza]: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/operators/README.md
+**Output Configuration - event**
+
+| Configuration | Default | Description
+| ------------ | ------------ | ------------
+| include_command_name | true | When set, includes the attribute `command.name` in the log event.
+| include_command_status | true | When set, includes the attribute `command.status` in the log event.
+| include_command_duration | true | When set, includes the attribute `command.duration` in the log event.
+| max_body_size | | When set, restricts the length of command output to a specified [ByteSize](#bytesize-parameters).
+
+**Output Configuration - log_entries**
+
+| Configuration | Default | Description
+| ------------ | ------------ | ------------
+| include_command_name | true | When set, includes the attribute `command.name` in the log event.
+| include_stream_name | true | When set, includes the attribute `command.stream.name` in the log event. Indicating `stdin` or `stderr` as the origin of the event.
+| max_log_size | | When set, restricts the length of a log entry to a specified [ByteSize](#bytesize-parameters).
+| encoding | utf-8 | Encoding to expect from the command output. Used to detect log entry boundaries.
+| multiline | | Used to override the default newline delimited log entries.
+| multiline.line_start_pattern | | Regex pattern for the beginning of a log entry. Mutualy exclusive with end pattern.
+| multiline.line_end_pattern | | Regex pattern for the ending of a log entry. Mutualy exclusive with start pattern.
+
+### Runtime Asset Configuration
+
+//todo(ck)
+
+## Additional Features
+
+### Time Parameters
+
+Time parameters are specified as sets of decimal numbers followed by a
+unit suffix. e.g. `60s`, `45m`, `2h30m40s`.
+
+### ByteSize Parameters
+
+ByteSize parameters can be specified either as an integer number or as a string
+starting with decimal numbers optionally followed by a common byte unit
+prefixes. e.g. `16kb`, `32MiB`, `1GB`.
+
+## Example configuration
+
+```yaml
+monitoringjob:
+  schedule:
+    interval: 1h
+  exec:
+    command: check_ntp_time
+    arguments:
+        - "-H"
+        - time.nist.gov
+    timeout: 8s
+  output:
```

Comment:
is there a reason that we need the output config in the receiver? Why not use the stanza operator in the receiver config like so:
```
receiver:
  monitoringjob:
    schedule:
       interval: 1h
    operators:
      - type: regex_parser
           ....
```

You may be able to use the [regex parser](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/operators/regex_parser.md), to parse/ include specific fields from the incoming log/event.

Reply:
@rnishtala-sumo I don't have particularly strong feelings on _where_ the stanza config lives. Maybe this will explain the choice to put it under `output`.

Take a look at output `type`. We want the receiver to run these jobs, but the output can be handled in two radically different ways. The default (and what I suspect the original intent in design was) type is event. Where the full output gets bundled up into a single log entry and emitted with a full context of the execution. The other option, `log_entires`, is meant to cover some popular existing sumo use cases (Script Sources) where the command output is streamed line by line.

We could probably do away with a nested _output_, but I thought it made more sense given the two output types.

Reply:
Recapping a conversation had out of band.

I set up the output using the existing upstream conventions in similar receivers like `filelogreceiver` including multi line processing, max log entry size. It is intended to be protective over collector resources.

An interesting experiment may be splitting individual log records in a "streaming" processor as Raj has suggested.